tests/unit/setup-rendering-context-test.js

```diff
@@ -115,4 +117,187 @@ module('setupRenderingContext', function(hooks) {
+    );
+
+    this.set('clicked', () => assert.ok(true, 'action was triggered'));
+    await this.render(hbs`{{x-foo clicked=clicked}}`);
+
+    assert.equal(this.element.textContent, 'Click me!', 'precond - component was rendered');
+    click(this.element.querySelector('button'));
+  });
+
+  test('can update a passed in argument with an <input>', async function(assert) {
+    this.owner.register(
+      'component:my-input',
+      TextField.extend({
+        value: null,
+      })
+    );
+
+    await this.render(hbs`{{my-input value=value}}`);
+
+    let input = this.element.querySelector('input');
+    input.value = '1';
+
+    fireEvent(input, 'change');
+    assert.equal(this.get('value'), '1');
+  });
+
+  test('it supports dom triggered focus events', async function(assert) {
+    this.owner.register(
+      'component:my-input',
+      TextField.extend({
+        init() {
+          this._super(...arguments);
+
+          this.set('value', 'init');
+        },
+        focusIn() {
+          this.set('value', 'focusin');
+        },
+        focusOut() {
+          this.set('value', 'focusout');
+        },
+      })
+    );
+    await this.render(hbs`{{my-input}}`);
+
+    let input = this.element.querySelector('input');
+    assert.equal(input.value, 'init');
+
+    focus(input);
+    assert.equal(input.value, 'focusin');
+
+    blur(input);
+    assert.equal(input.value, 'focusout');
+  });
+
+  test('two way bound arguments are updated', async function(assert) {
+    this.owner.register(
+      'component:my-component',
+      Component.extend({
+        actions: {
+          clicked() {
+            this.set('foo', 'updated!');
+          },
+        },
+      })
+    );
+    this.owner.register(
+      'template:components/my-component',
+      hbs`<button {{action 'clicked'}}>{{foo}}</button>`
+    );
+
+    this.set('foo', 'original');
+    await this.render(hbs`{{my-component foo=foo}}`);
+    assert.equal(this.element.textContent, 'original', 'value after initial render');
+
+    click(this.element.querySelector('button'));
+    assert.equal(this.element.textContent, 'updated!', 'value after updating');
+    assert.equal(this.get('foo'), 'updated!');
+  });
+
+  test('two way bound arguments are available after clearRender is called', async function(assert) {
+    this.owner.register(
+      'component:my-component',
+      Component.extend({
+        actions: {
+          clicked() {
+            this.set('foo', 'updated!');
+            this.set('bar', 'updated bar!');
+          },
+        },
+      })
+    );
+    this.owner.register(
+      'template:components/my-component',
+      hbs`<button {{action 'clicked'}}>{{foo}}</button>`
+    );
+
+    await this.render(hbs`{{my-component foo=foo bar=bar}}`);
+    click(this.element.querySelector('button'));
+
+    await this.clearRender();
+
+    assert.equal(this.get('foo'), 'updated!');
+    assert.equal(this.get('bar'), 'updated bar!');
```

Comment:
why two properties? because one of them is rendered and the other is not? might need a comment 🤔

Reply:
Kk, sounds good. The idea was to confirm that two way binding works regardless of if the property was rendered.